src/pymor/models/iosys.py

```diff
@@ -1500,6 +1500,161 @@ def __init__(self, A, Ad, tau, B, C, D=None, E=None, cont_time=True,
         self.tau = tau
         self.q = len(Ad)
 
+    def __add__(self, other):
+        """Add an |LTIModel|, |SecondOrderModel| or |LinearDelayModel|."""
+        assert self.cont_time == other.cont_time
+        assert self.input_space == other.input_space
+        assert self.output_space == other.output_space
+
+        if isinstance(other, SecondOrderModel):
+            other = other.to_lti()
+
+        if isinstance(other, LTIModel):
+            Ad = tuple(BlockDiagonalOperator([op, ZeroOperator(other.solution_space, other.solution_space)])
+                       for op in self.Ad)
+            tau = self.tau
+        elif isinstance(other, LinearDelayModel):
+            tau = tuple(set(self.tau).union(set(other.tau)))
+            Ad = [None for _ in tau]
+            for i, taui in enumerate(tau):
+                if taui in self.tau and taui in other.tau:
+                    Ad[i] = BlockDiagonalOperator([self.Ad[self.tau.index(taui)],
+                                                   other.Ad[other.tau.index(taui)]])
+                elif taui in self.tau:
+                    Ad[i] = BlockDiagonalOperator([self.Ad[self.tau.index(taui)],
+                                                   ZeroOperator(other.solution_space, other.solution_space)])
+                else:
+                    Ad[i] = BlockDiagonalOperator([ZeroOperator(self.solution_space, self.solution_space),
+                                                   other.Ad[other.tau.index(taui)]])
+            Ad = tuple(Ad)
+        else:
+            return NotImplemented
+
+        E = BlockDiagonalOperator([self.E, other.E])
+        A = BlockDiagonalOperator([self.A, other.A])
+        B = BlockColumnOperator([self.B, other.B])
+        C = BlockRowOperator([self.C, other.C])
+        D = self.D + other.D
+        return self.with_(E=E, A=A, Ad=Ad, tau=tau, B=B, C=C, D=D)
+
+    def __radd__(self, other):
+        """Add to an |LTIModel| or a |SecondOrderModel|."""
+        if isinstance(other, LTIModel):
+            return self + other
+        elif isinstance(other, SecondOrderModel):
+            return self + other.to_lti()
+        else:
+            raise NotImplementedError
```

Comment:
Shouldn't this be `return NotImplemented`?

Reply:
Right. Fixed it now also in `SecondOrderModel`.